amaranth/build/run.py

```diff
@@ -138,6 +138,7 @@ def execute_local_docker(self, image, *, root="build", docker_args=[]):
         subprocess.check_call([
             "docker", "run", *docker_args,
             "--rm", # remove the container after running
+           "-it" # open container with interactive terminal session, allowing it to properly catch ctrl-C exit from user.
```

Comment:
Can this not cause more problems? What happens if you run Amaranth itself from a script? I vaguely recall removing `-it` from the option list before.